go/vt/sqlparser/ast.go

```diff
@@ -97,6 +97,26 @@ func ParseNext(tokenizer *Tokenizer) (Statement, error) {
 	return tokenizer.ParseTree, nil
 }
 
+// SplitStatement returns the first sql statement up to either a ; or EOF
+// and the remainder from the given buffer
+func SplitStatement(blob string) (string, string, error) {
```

Comment:
I think you can just use sqlparser.ParseNext instead.

Reply:
I started there, but I didn't want to actually bother parsing the tree since I wanted to distinguish between any scanner errors and parser errors.

I can probably switch back but this seemed pretty straightforward.